.github/workflows/build-test.yml

```diff
@@ -0,0 +1,53 @@
+name: Build
+
+on:
+  push:
+    tags:
+      - '*.*.*.*'
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+      tags:
+        description: 'Test scenario tags'
+
+jobs:
+  build-n-publish:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    
+    - name: Check out code/repo into the directory
+      uses: actions/checkout@v2
+
+    - name: Get the Ref
+      id: get-ref
+      uses: ankitvgupta/ref-to-tag-action@master
+      with:
+        ref: ${{ github.ref }}
+        head_ref: ${{ github.head_ref }}
+    
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.x
+        
+    - name: Install pypa and depencies
+      run: |
+        python -m pip install --upgrade pip --user
```

Comment:
I think it would be more safe if you'll use `python3` instead of `python`. The latter usually refer to python 2. The same goes to `pip3` vs `pip`.